### PR TITLE
[Merged by Bors] - chore: `ValuedCsp` and `ValuedCsp.Instance` changed from `def` to `abbrev`

### PR DESCRIPTION
**Pull request by madvorak**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Combinatorics/Optimization/ValuedCSP.lean [hunk 1]
```diff
@@ -53,6 +53,7 @@ def ValuedCsp.Term.evalSolution {Γ : ValuedCsp D C} {ι : Type*}
   t.f (x ∘ t.app)
 
 /-- A valued CSP instance over the template `Γ` with variables indexed by `ι`.-/
+@[reducible]
 def ValuedCsp.Instance (Γ : ValuedCsp D C) (ι : Type*) : Type _ :=
```
Comment (eric-wieser) [inline Mathlib/Combinatorics/Optimization/ValuedCSP.lean:57]: Instead of reducible, you can change `def` to `abbrev`

Comment (madvorak) [reply Mathlib/Combinatorics/Optimization/ValuedCSP.lean:57]: Are there any general recommendations about when to use `abbrev`? For comparison, `ValuedCsp` is `reducible` `def`.

Comment (eric-wieser) [reply Mathlib/Combinatorics/Optimization/ValuedCSP.lean:57]: I think `@[reducible] def` was a lean3-ism cause by `abbrev`(`iation`) generating bad docs and no equation lemmas. That flaw is absent in Lean 4.

Comment (madvorak) [reply Mathlib/Combinatorics/Optimization/ValuedCSP.lean:57]: Should both `ValuedCsp` and `ValuedCsp.Instance` be `abbrev` then? Will dot notation work exactly the same?

Comment (eric-wieser) [reply Mathlib/Combinatorics/Optimization/ValuedCSP.lean:57]: Yes, basically nothing will care.

**Review by eric-wieser — APPROVED** [review]
bors merge

Thanks!

**Pull request closed without merging**